### ISSUE TITLE
is_call with quosures is not consistent with call_name

### ISSUE THREAD

**Issue by asardaes** [r-lib/rlang#796]
I'm not sure if it's related to #670 

```r
colon_quo <- rlang::quo(1:2)

is_call(colon_quo, ":")
# FALSE

call_name(colon_quo)
# ":"

is_call(quo_get_expr(colon_quo), ":")
# TRUE


tidyselect_quo <- quo(everything())

is_call(tidyselect_quo, names(tidyselect::vars_select_helpers))
# FALSE

call_name(tidyselect_quo)
# "everything"

is_call(quo_get_expr(tidyselect_quo), names(tidyselect::vars_select_helpers))
# TRUE
```

**Comment by lionel-** [comment]
yup duplicate of #670.

**Comment by asardaes** [comment]
I see that the linked issue was closed, but my example here still shows the discrepancy.

**Comment by lionel-** [comment]
Unlike `call_name()` and `is_call_simple()`, `is_call()` is a low level function. So either:

1. The current behaviour is correct. `is_call(quo, "~")` returns `TRUE` now but maybe not in the future when the internal representation of quosures changes.

2. Quosures should not be treated as calls and `is_call(quo)` should _always_ return `FALSE`.

Some years ago I leaned towards 2 because I conceived of quosures as their own new type. However changing this broke purrr which lead me to revert the change. Now I'm not so sure and I lean towards 1. In the future plans for quosures, they are still implemented as calls: https://github.com/r-lib/rlang/issues/1145. I find it increasingly appropriate for a native quosure to be implemented as a LANGSXP.

The bottom line is that the current behaviour is reasonable and maybe even correct :)

Maybe `is_call_simple()` in https://rlang.r-lib.org/reference/call_name.html will help you though.

**Comment by asardaes** [comment]
Then may I suggest adjusting documentation? In `call_name`, `call` "Can be a call or a quosure that wraps a call", which seems to hint that it will ignore the call to `~` if it's a quosure. In `is_call`, `x` is "An object to test. If a formula, the right-hand side is extracted". If quosures count as formulas right now, that description makes me thing that there should be no discrepancy, and if quosures will move away from formulas in the future, a heads-up would be nice, something like "quosures are not formulas and have special semantics".

**Comment by lionel-** [comment]
Thanks, this must have been a left-over from a very early version of rlang. The description for formulas is now updated in https://github.com/r-lib/rlang/commit/9b112ead55c6a405e5557528a9143942394331d9.
